### ISSUE TITLE
[Feature] Possibility to render description and properties for different elements on views

### ISSUE THREAD

**Issue by vgotra** [archimatetool/archi#171]
Please a possibility to render description and properties for different elements on views (like at attached image).
![element](https://cloud.githubusercontent.com/assets/896830/21158015/4ecb9860-c184-11e6-85a6-814c7e02dd9b.png)

**Comment by jbsarrodie** [comment]
Hi,

Phil and I had a recent discussion about this idea, but this raised several UX questions:
* Does this has to be set on a per view basis (and not a one off option associated with the concept)?
* If someone wants to select the list of properties to show, then is this list configured on a per view basis?
* How do we show properties? Only name? value? both?

If we are able to define a good (but simple) behavior, then this might be done after Archi 4.0 release.

Regards

**Comment by vgotra** [comment]
Hi

> Does this has to be set on a per view basis (and not a one off option associated with the concept)?   
> If someone wants to select the list of properties to show, then is this list configured on a per view basis?

It can be configurable - in properties of element can be added column with checkboxes to enable default behavior of element (display/not display some properties). And then on some view it can be overridden.
Per view implementation can be much more simple to implement - but it will not be very flexible (in this case person should configure every view. Case with default behavior per element can save time for every view.

> How do we show properties? Only name? value? both?

Both - this will help to easy understand what we have at view without export to check generated file.

Thanks

**Comment by jbsarrodie** [comment]
Hi,

_Per view implementation can be much more simple to implement_

In fact that's the opposite. Properties are linked to concepts in the model, not the view, so this new attribute (show the property) could easily be added on the property (making this a model wide state) but it's very difficult to define a per view state (this would require a new kind of information that links not only the element to a view but each properties, making sure that if a property is removed all links are too).

_Both - this will help to easy understand what we have at view without export to check generated file._

That might be the best option for your use case but that's not that simple. Another use case it to do kind of data modelling. In this case you would use property name for the attribut name, and property value for the textual description of the attribute. This means that only names should be seen on view (value would just be used for documentation purposes).

So to summarize, not so simple to find a simple implementation that covers all these use cases, So we have to think twice about a potential first step that would not have to be fully redesigned in the future.

Regards,

**Comment by theunsfvo** [comment]
This would be very helpful as well, at least the description part of the element to show descriptions next to elements for illustration purposes... instead of having to create a note and copying the description text into it for every element...

Even if it is a right click option on an element or maybe a toggle/button on the properties view to show it in the view next to the element?

It can be just a simple "create note with description text" function place close to the mouse cursor on the view...

**Comment by jbsarrodie** [comment]
>It can be just a simple "create note with description text" function place close to the mouse cursor on the view...

This is another thing IMHO, but maybe easier to implement: One could add a checkbox on a note to replace content by the description attribute of the concept connected to the note. Some obvious safe guards should be added though.

Re the original discussion, a first MVP could be to simply add a visual atributte (ie. attribute related to the context of the view, not the element itself) to add **all** properties keys and/or **all** properties values (not a one by one selection.

**Comment by theunsfvo** [comment]
Perhaps we should list it as a separate feature request then?

That alone (auto description note generation) will already add a lot of value, even though other property rendering would also be very useful...

**Comment by FlightControl-User** [comment]
Is there any work done on this item till today?

**Comment by jbsarrodie** [comment]
> Is there any work done on this item till today?

Yes and no ;-)

You can achieve the same result by using [Hervé's specialization plugin](https://github.com/archi-contribs/specialization-plugin).

Phil and I are planning to add this feature (label customization) natively in Archi in 2020. See our [roadmap](https://www.archimatetool.com/roadmap/).

**Comment by Phillipus** [comment]
This is implemented in #576 so closing this.
